app/src/marketplace/containers/GlobalInfoWatcher/index.jsx

```diff
@@ -123,6 +125,28 @@ export const GlobalInfoWatcher = ({ children }: Props) => {
         }
     }, [dispatch, balancePoll, username])
 
+    // Poll network
+    const handleNetwork = useCallback((nextNetworkId: NumberString) => {
+        setNetworkId((prevNetworkId) => {
+            // refresh page if network changed
+            if (!!prevNetworkId && !!nextNetworkId && prevNetworkId !== nextNetworkId) {
+                window.location.reload()
+            }
+
+            return nextNetworkId
+        })
+    }, [])
+
+    useEffect(() => {
+        Web3Poller.subscribe(Web3Poller.events.NETWORK, handleNetwork)
+        Web3Poller.subscribe(Web3Poller.events.NETWORK_ERROR, handleNetwork)
+
+        return () => {
+            Web3Poller.unsubscribe(Web3Poller.events.NETWORK, handleNetwork)
+            Web3Poller.unsubscribe(Web3Poller.events.NETWORK_ERROR, handleNetwork)
+        }
+    }, [handleNetwork])
```

Comment:
I wouldn't say we gotta go with `useState` for this one. Check out my idea:

```suggestion
    useEffect(() => {
        const currentNetworkId = undefined

        const onNetworkChange = (networkId) => {
            if (currentNetworkId && networkId && currentNetworkId !== networkId) {
                window.location.reload()
            }
        }

        Web3Poller.subscribe(Web3Poller.events.NETWORK, onNetworkChange)
        Web3Poller.subscribe(Web3Poller.events.NETWORK_ERROR, onNetworkChange)

        return () => {
            Web3Poller.unsubscribe(Web3Poller.events.NETWORK, onNetworkChange)
            Web3Poller.unsubscribe(Web3Poller.events.NETWORK_ERROR, onNetworkChange)
        }
    }, [])
```

Reply:
Ok, maybe that's better so we don't need the `useState` 👍